tests/flux-v2-image-automation.sh

```diff
@@ -56,7 +62,54 @@ for FILE_LOCATION in $(echo ${FILE_LOCATIONS}); do
                 echo "Non whitelisted pattern found in ImagePolicy: $IMAGE_POLICY it should be ^prod-[a-f0-9]+-(?P<ts>[0-9]+)" && exit 1
             fi
         done
+    done
 
+    ##############################################################################################################
+    # Print success if ALL image policies are clean
+    ##############################################################################################################
+    printf "\n\n########## Image Policy documents checked and passing ########## \n\n"
+
+    ##############################################################################################################
+    # This section compiles a list of files that contains `image:` and stores them in an array HELMRELEASES
+    # Only files that follow these rules will be added to the array:
+    # - Contains the `image:` string
+    # - Is HelmRelease type document
+    ##############################################################################################################
+
+    HELMRELEASES=()
+    for FILE in $(grep -lr "image:" $FILE_LOCATION | grep -Ev "$EXCLUSIONS" ); do
+        while read -r doc; do
+            if [ "$doc" == "HelmRelease" ]; then
+                IFS=$'\n'
+                HELMRELEASES+=("$FILE")
+            fi
+        done < <(yq eval '.kind' $FILE)
     done
 
+    OUTPUTFILE="images.yaml"
+
+    for dir in $DIRECTORIES; do
+        exclude=false
+        for EXCLUDED_PATH in "${HELMRELEASES[@]}"; do
+        if [[ -f "$dir/$(basename $EXCLUDED_PATH)" ]]; then
```

Comment:
Not sure what this If is doing.

The contents of HELMRELEASE look like:
```
apps/darts-modernisation/darts-portal/darts-portal.yaml
apps/darts-modernisation/darts-stub-services/darts-stub-services.yaml
```

So this if statement will result in `apps/darts-stub-services.yaml` which doesnt exist but that would be the case for all of the files found in HELMRELEASE as none exist in the top-level `apps` folder.